docker/proof-server-chart/templates/proof-server-deployment.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "proof-server.fullname" . }}-deployment
+  labels:
+  {{- include "proof-server.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.deployment.replicas }}
+  selector:
+    matchLabels:
+      app: proof-server
+    {{- include "proof-server.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app: proof-server
+      {{- include "proof-server.selectorLabels" . | nindent 8 }}
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.deployment.proofServer.terminationGracePeriodSeconds }}
+      nodeSelector:
+        {{- toYaml .Values.deployment.proofServer.nodeSelector | nindent 8 }}
+      containers:
+      - command:
+        - sh
+        - -c
+        - /app/proof_server --mode "single" --addr ${CONTAINER_ADDR}:${CONTAINER_PORT}
+        env:
+        - name: RUST_LOG
+          value: {{ quote .Values.deployment.proofServer.env.rustLog }}
+        - name: RUST_MIN_STACK
+          value: {{ quote .Values.deployment.proofServer.env.rustMinStack }}
+        - name: SHARD_SIZE
+          value: {{ quote .Values.deployment.proofServer.env.shardSize }}
+        - name: SHARD_BATCH_SIZE
+          value: {{ quote .Values.deployment.proofServer.env.shardBatchSize }}
+        - name: SHARD_CHUNKING_MULTIPLIER
+          value: {{ quote .Values.deployment.proofServer.env.shardChunkingMultiplier }}
+        - name: RECONSTRUCT_COMMITMENTS
+          value: {{ quote .Values.deployment.proofServer.env.reconstructCommitments }}
+        - name: CONTAINER_ADDR
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: CONTAINER_PORT
+          value: {{ quote .Values.deployment.proofServer.env.containerPort }}
+        - name: KUBERNETES_CLUSTER_DOMAIN
+          value: {{ quote .Values.kubernetesClusterDomain }}
+        image: {{ .Values.deployment.proofServer.image.repository }}:{{ .Values.deployment.proofServer.image.tag
+          | default .Chart.AppVersion }}
+        imagePullPolicy: {{ .Values.deployment.proofServer.imagePullPolicy }}
+        livenessProbe:
+          httpGet:
+            path: /health
+            port: 8080
+          initialDelaySeconds: 10
+          periodSeconds: 5
+        readinessProbe:
+          httpGet:
+            path: /ready
+            port: 8080
+          initialDelaySeconds: 15
+          periodSeconds: 10
+        name: proof-server
+        ports:
+        - containerPort: 8080
+        resources: {}
```

Comment:
The resources field being empty makes me wonder: how would things work if we wanted to migrate out of the nodeSelector of EKS?

Would a minimum usable amount of hardware be useful here? @samuelburnham may have ideas.

```
resources:
  requests:
    memory: "128Gi"
    cpu: "16"
  limits:
    memory: "128Gi"
    cpu: "32"
```

Reply:
I think that we want to use `r7iz.metal-32xl ` anyway as we base our benchmark on these. If some users want another type of hardware they can always change the configuration. 

To be 100% honest I'm not even sure how to properly test this workflow outside of an `eksctl create` then `helm install` so making this change might take some time to validate.

I created #234 for it.